src/ucp/wireup/wireup_cm.c

```diff
@@ -0,0 +1,272 @@
+/**
+* Copyright (C) Mellanox Technologies Ltd. 2019.  ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#ifdef HAVE_CONFIG_H
+#  include "config.h"
+#endif
+
+#include "wireup_cm.h"
+#include <ucp/core/ucp_ep.inl>
+#include <ucp/wireup/wireup.h>
+#include <ucp/wireup/wireup_ep.h>
+
+
+static unsigned
+ucp_cm_ep_init_flags(const ucp_worker_h worker, const ucp_ep_params_t *params)
+{
+    if (!ucp_worker_sockaddr_is_cm_proto(worker)) {
+        return 0;
+    }
+
+    if (params->field_mask & UCP_EP_PARAM_FIELD_SOCK_ADDR) {
+        return UCP_EP_INIT_CM_WIREUP_CLIENT;
+    }
+
+    if (params->field_mask & UCP_EP_PARAM_FIELD_CONN_REQUEST) {
+        return UCP_EP_INIT_CM_WIREUP_SERVER;
+    }
+
+    return 0;
```

Comment:
should we fail if getting here?

Reply:
No, this will be normal flow for common code